tests/h/services/search_index/_queue_test.py

```diff
@@ -0,0 +1,255 @@
+
+    def test_if_there_are_multiple_jobs_with_the_same_annotation_id(
+        self, annotation_ids, batch_indexer, caplog, queue
+    ):
+        queue.add_all(
+            [annotation_ids[0], annotation_ids[0], annotation_ids[0]],
+            tag="test",
+            scheduled_at=MINUS_FIVE_MINUTES,
+        )
+
+        queue.sync()
+
+        # It only syncs the annotation to Elasticsearch once.
+        assert caplog.record_tuples == [
+            (Any(), Any(), "Syncing 1 annotations that are missing from Elasticsearch")
+        ]
+        batch_indexer.index.assert_called_once_with(
+            Any.list.containing([annotation_ids[0]]).only()
+        )
+
+    def test_deleting_multiple_jobs_with_the_same_annotation_id(
+        self, annotations, batch_indexer, caplog, db_session, index, queue
+    ):
+        queue.add_all(
+            [annotations[0].id, annotations[0].id, annotations[0].id],
+            tag="test",
+            scheduled_at=MINUS_FIVE_MINUTES,
+        )
+        index([annotations[0]])
+
+        queue.sync()
+
+        assert caplog.record_tuples == [
+            (Any(), Any(), "Deleting 3 successfully completed jobs from the queue")
+        ]
+        assert db_session.query(Job).all() == []
+        batch_indexer.index.assert_not_called()
+
+    @pytest.fixture
+    def annotations(self, factories):
+        return factories.Annotation.create_batch(size=20)
+
+    @pytest.fixture
+    def all_annotation_ids(self, annotations):
+        return [annotation.id for annotation in annotations]
+
+    @pytest.fixture
+    def annotation_ids(self, all_annotation_ids):
+        return all_annotation_ids[:LIMIT]
+
+    @pytest.fixture
+    def caplog(self, caplog):
+        # Filter out log messages from any other module.
+        caplog.set_level(logging.CRITICAL + 1)
+        # Filter in log messages from the module under test only.
+        caplog.set_level(logging.INFO, "h.services.search_index._queue")
+        return caplog
+
+    @pytest.fixture
+    def index(self, es_client, moderation_service, nipsa_service, pyramid_request):
+        """A function for adding annotations to Elasticsearch."""
+
+        def index(annotations, updated=None):
+            """Add `annotations` to the Elasticsearch index."""
+            for annotation in annotations:
+                body = AnnotationSearchIndexPresenter(
+                    annotation, pyramid_request
+                ).asdict()
+
+                if updated is not None:
+                    body["updated"] = updated
+
+                es_client.conn.index(
+                    index=es_client.index,
+                    doc_type=es_client.mapping_type,
+                    body=body,
+                    id=annotation.id,
+                )
+            es_client.conn.indices.refresh(index=es_client.index)
+
+        return index
+
+
+LIMIT = 10
+
+
+ONE_WEEK = datetime_.timedelta(weeks=1)
+
+
+MINUS_FIVE_MINUTES = datetime_.timedelta(minutes=-5)
```

Comment:
I think fixtures go at the bottom, but constants generally go to the top. The double spacing is a touch odd too.